changes/watch_darwin.go

```diff
@@ -0,0 +1,84 @@
+// heavily inspired by https://github.com/fsnotify/go.fsevents
```

Comment:
I updated the comment to point to the other repo as this is largely influenced by the ideas in this wrapper but is not an outright copy. Thanks!